multipledispatch/tests/test_dispatcher.py

```diff
@@ -258,3 +261,139 @@ def _(a):
         raise MDNotImplementedError()
 
     assert raises(NotImplementedError, lambda: f(1.0))
+
+
+def test_vararg_not_last_element_of_signature():
+    f = Dispatcher('f')
+    assert raises(TypeError, lambda: f.register([float], str)(lambda: None))
+
+
+def test_vararg_has_multiple_elements():
+    f = Dispatcher('f')
+    assert raises(TypeError, lambda: f.register([float, str])(lambda: None))
+
+
+def test_vararg_dispatch_simple():
+    f = Dispatcher('f')
+
+    @f.register([float])
+    def _1(*args):
+        return args
+
+    assert f(1.0) == (1.0,)
+    assert f(1.0, 2.0, 3.0) == (1.0, 2.0, 3.0)
+
+
+def test_vararg_dispatch_ambiguity():
```

Comment:
we might want to add a tests where the ambiguity is in the variadic itself like:
```
In [3]: @f.register(object, [int])
   ...: def _(*args):
   ...:     pass
   ...: 

In [4]: @f.register(int, [object])
   ...: def _(*args):
   ...:     pass
   ...: 

```

Reply:
Agreed.

Reply:
Added